src/IDriver.php

```diff
@@ -19,33 +17,157 @@ public function graph($query,$includeProperties=array());
      * @return mixed
      */
     public function describe($query);
+
+    /**
+     * Return (DESCRIBE) the concise bound description of a resource
+     * @param $resource string uri resource you'd like to describe
+     * @param null $context string uri of the context, or named graph, you'd like to describe from
+     * @return ExtendedGraph
+     */
     public function describeResource($resource,$context=null);
+
+    /**
+     * Return (DESCRIBE) the concise bound descriptions of a bunch of resources
+     * @param array $resources uris of resources you'd like to describe
+     * @param null $context string uri of the context, or named graph, you'd like to describe from
+     * @return ExtendedGraph
+     */
     public function describeResources(Array $resources,$context=null);
 
+    /**
+     * Get a view of a given type for a given resource
+     * @param $resource string uri of the resource you'd like the view for
+     * @param $viewType string type of view
+     * @return ExtendedGraph
+     */
     public function getViewForResource($resource,$viewType);
+
+    /**
+     * Get views for multiple resources in one graph
+     * @param array $resources uris of resources you'd like to describe
+     * @param $viewType string type of view
+     * @return ExtendedGraph
+     */
     public function getViewForResources(Array $resources,$viewType);
+
+    /**
+     * Get views based on a pattern-match $filter
+     * @param array $filter pattern to match to select views
+     * @param $viewType string type of view
+     * @return ExtendedGraph
+     */
     public function getViews(Array $filter,$viewType);
 
+    /**
+     * Returns the etag of a resource, useful for caching
+     */
     public function getETag($resource,$context=null);
 
-    // tabular functions
-
-    public function select($query,$fields,$sortBy=null,$limit=null,$context=null);
+    /**
+     * Select data in a tabular format
+     * @param array $filter pattern to match to select views
+     * @param $fields
+     * @param null $sortBy
```

Comment:
Not to trod on your 30%-er with 90%-er comments, but these are really handy and I don't want them to get lost in the actual functionality of this branch: make sure the param types of these aren't null.